### PR TITLE
Add testing nvJpeg with available training data and add fallback to host

### PR DESCRIPTION
**Pull request by JanuszL**
for other than jpeg images

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

**Comment by JanuszL** [comment]
Tested internally 33513.

File: dali/pipeline/operators/decoder/nvjpeg_decoder.h [hunk 1]
```diff
@@ -161,25 +162,33 @@ class nvJPEGDecoder : public Operator<MixedBackend> {
...
 
       // Get necessary image information
-      NVJPEG_CALL(nvjpegGetImageInfo(handle_,
+      nvjpegStatus_t ret = nvjpegGetImageInfo(handle_,
                                      static_cast<const unsigned char*>(data), in_size,
                                      &info.c, &info.subsampling,
-                                     info.widths, info.heights));
+                                     info.widths, info.heights);
+      // Fallback for png
+      if (ret == NVJPEG_STATUS_BAD_JPEG && CheckIsPNG(static_cast<const uint8*>(data), in_size)) {
```
Comment (cliffwoolley) [inline dali/pipeline/operators/decoder/nvjpeg_decoder.h]: What if it's not a JPEG but also not a PNG?  Is there something more general we could do here?

Comment (ptrendx) [reply dali/pipeline/operators/decoder/nvjpeg_decoder.h]: Not really unfortunately (besides ripping something from ImageMagick or something like that) - basically codes like OpenCV do not advertise functionality for getting info from image header without decoding the image too.

Comment (ptrendx) [reply dali/pipeline/operators/decoder/nvjpeg_decoder.h]: There is stuff like this: https://github.com/scardine/image_size the logic from which we could use to expand the capabilities here (which would be much easier than searching for that functionality in OpenCV/ImageMagick source)

**Comment by JanuszL** [comment]
@cliffwoolley all test.sh looks similar, and yes it is a good idea to refactor them. It is yet another poin in ToDo list.

**Pull request merged**